File: scripts/01-download_data.R
#### Preamble ####
# Purpose: Downloads and saves the data from [...UPDATE THIS...]
# Author: Mandy He
# Date: 23 September 2024 
# Contact: mandyxy.he@mail.utoronto.ca
# License: MIT
# Pre-requisites: [...UPDATE THIS...]
# Any other information needed? [...UPDATE THIS...]


#### Workspace setup ####
library(opendatatoronto)
library(tidyverse)
library(dplyr)


#### Download data ####

# get package
package <- show_package("deaths-of-shelter-residents")
package

# get all resources for this package
resources <- list_package_resources("deaths-of-shelter-residents")

# identify datastore resources; by default, Toronto Open Data sets datastore resource format to CSV for non-geospatial and GeoJSON for geospatial resources
datastore_resources <- filter(resources, tolower(format) %in% c('csv', 'geojson'))

# load the Deaths of Shelter Residents data
Num_death_data <- filter(datastore_resources, row_number()==1) %>% get_resource()


# Deaths of Shelter Residents Mean Age
Avg_death_age_data <- filter(datastore_resources, row_number()==2) %>% get_resource()


#### Save data ####
write_csv(Num_death_data, "data/raw_data/Num_death_raw_data.csv") 
write_csv(Avg_death_age_data, "data/raw_data/raw_age_data.csv") 

         

File: scripts/01-download_data.py
#### Preamble ####
# Purpose: Downloads and saves the data from Open Data Toronto
# Author: Mandy He
# Date: 23 September 2024
# Contact: mandyxy.he@mail.utoronto.ca
# License: N/A
# Pre-requisites: N/A

	
import requests
from io import StringIO
import pandas as pd
 
# Toronto Open Data is stored in a CKAN instance. It's APIs are documented here:
# https://docs.ckan.org/en/latest/api/
 
# To hit our API, you'll be making requests to:
base_url = "https://ckan0.cf.opendata.inter.prod-toronto.ca"
 
# Datasets are called "packages". Each package can contain many "resources"
# To retrieve the metadata for this package and its resources, use the package name in this page's URL:
url = base_url + "/api/3/action/package_show"
params = { "id": "deaths-of-shelter-residents"}
package = requests.get(url, params = params).json()
 
dataframe = []
# To get resource data:
for idx, resource in enumerate(package["result"]["resources"]):
 
       # for datastore_active resources:
       if resource["datastore_active"]:
 
           # To get all records in CSV format:
           url = base_url + "/datastore/dump/" + resource["id"]
           resource_dump_data = requests.get(url)
           data = pd.read_csv(StringIO(resource_dump_data.text))

           dataframe.append(data)
 
Num_death_data = dataframe[0]
Avg_death_age_data = dataframe[1]


#### Save data ####
Num_death_data.to_csv("data/raw_data/Num_death_raw_data.csv", index = False)
Avg_death_age_data.to_csv("data/raw_data/raw_age_data.csv", index = False)
